CONTRIBUTING.md

```diff
@@ -0,0 +1,170 @@
+# Contributing to PivorakWebApp
+:+1::tada: Greetings, contributor! From the #pivorak Ruby usergroup we want to say thank you for joining us on this path to community development. We appreaciate your effort, every change counts!
+ :tada::+1:
+
+The following is a set of guidelines for contributing to PivorakWebApp. We've tried our best to ease this process for you, so please read it carefully.
+
+#### Table Of Contents
+- [Code of conduct](https://pivorak.com/code-of-conduct)
+- [How to setup project](#how-to-setup-project)
+    - [Option 1: Without docker](#option-1-without-docker)
+    - [Option 2: With docker](#option-2-with-docker)
+- [How to report a bug](#how-to-report-a-bug)
+- [What should I know before I get started?](#what-should-i-know-before-i-get-started)
+    - [Git Branches](#git-branches)
+    - [Design Decisions](#design-decisions)
+
+# How to setup project
+## Option 1: Without docker
+#### Step 1: Clone the repo
+```sh
+$ git clone https://github.com/pivorakmeetup/pivorak-web-app.git
+```
+#### Step 2: Install dependencies
+- Install ruby 2.3.1
+```sh
+$ cd pivorak-web-app
+$ rvm install ruby-2.3.1
+```
+- Install gems
+```sh
+$ rvm use ruby-2.3.1@pivorak --create
+$ bundle
+```
+- Setup database
+    **Notice:** Before database setup you should install *Redis*. Look install instructions here:
+    - for Ubuntu: https://goo.gl/PbG00W
+    - for MacOS: https://goo.gl/W8xQMJ
+
+    **Notice:**  You should have PostgreSQL >= 9.4
+    To check your version
+```sh
+$ psql --version
+```
+
+```sh
+$ cp config/database.yml.example config/database.yml
+$ rails db:setup
+$ rails s
+```
+
+**Notice:**  If you get an error messages during the `rails db:setup` like this:
+```sh
+'require': cannot load such file -- bundler (LoadError)
+```
+then try to run `gem pristine --all` and after that `rails db:setup`.
+
+
+## Option 2: With docker
+#### Step 1: Install [Docker](https://docs.docker.com/install/) and [Docker Compose](https://docs.docker.com/compose/install/)
+#### Step 2: Clone the repo
+```sh
+$ git clone https://github.com/pivorakmeetup/pivorak-web-app.git
+```
+#### Step 3: Clone database config
+```sh
+$ cd pivorak-web-app
+$ cp config/database.yml.example config/database.yml
+```
+#### Step 4: Build project
+```sh
+$ docker-compose build
+$ docker-compose run runner bundle install
+```
+
+**Notice:** How to deal with docker?
+
+**Test environment:**
+To run tests, first setup test database
+```sh
+$ docker-compose run -e RAILS_ENV=test runner bundle exec rails db:setup
+```
+and then run test suite
+
+```sh
+$ docker-compose run runner bundle exec rspec
+```
+
+**Development environment:**
+To run the app in development, first prepare development database
+```sh
+$ docker-compose run runner bundle exec rails db:setup
+```
+and then execute
+```sh
+$ docker-compose up rails
+```
+
+# How to contribute
```

Comment:
Add paragraph related to **How to contribute during Hacktoberfest** (or whatever the name is) which will be almost identical to How to contribute except few details like:
1. Check issues with the tag...
2. Fork project  - without adding a note about permissions 
3. In case of emergency question contact with .....